reconcile/utils/terrascript_client.py

```diff
@@ -3299,6 +3299,32 @@ def populate_tf_resource_elasticsearch(self, resource, namespace_info):
         if self._multiregion_account_(account):
             es_values['provider'] = 'aws.' + region
 
+        advanced_security_options = values.get('advanced_security_options', {})
+        if advanced_security_options:
+            master_user_options = advanced_security_options.pop(
+                'master_user_options', {})
+
+            es_values['advanced_security_options'] = advanced_security_options
+
+            if master_user_options:
+                master_user_secret = master_user_options['master_user_secret']
+                secret_data = self.secret_reader.read_all(master_user_secret)
+
+                required_keys = ['master_user_name', 'master_user_password']
+                found = 0
+                for key in required_keys:
+                    if key in secret_data:
+                        found += 1
+
+                if len(required_keys) != found:
```

Comment:
sure, less variables... :)

Reply:
hmmm, your proposal not working as I expected. This should not succeed:
```
>>> secret_data = {'master_user_name': "u", 'master_user_password': "p"}
>>> required_keys = {'master_user_name', 'master_user_password'}
>>> not required_keys < secret_data.keys()
True
```
But mine is not working too. It would appear that it is:
```
>>> secret_data.keys() < required_keys
False
```
But it actually fails as this should return True
```
>>> secret_data = {'u': "u", 'p': "p"}
>>> secret_data.keys() < required_keys
False
```
I guess it is actually comparing lengths.

So if we want to continue with the `required_keys` set approach, we could use intersections
```
>>> secret_data = {'master_user_name': "u", 'master_user_password': "p"}
>>> required_keys = {'master_user_name', 'master_user_password'}
>>> secret_data.keys() & required_keys != required_keys
False
>>> secret_data = {'u': "u", 'p': "p"}
>>> secret_data.keys() & required_keys != required_keys
True
>>> secret_data = {'master_user_name': "u"}
>>> secret_data.keys() & required_keys != required_keys
True
```

Reply:
Fixed in 9e087c6

Reply:
It was easier, for sure. A simple comparison would have sufficed:
```
>>> secret_data = {'master_user_name': "u", 'master_user_password': "p", "this": "should not be here"}
>>> required_keys = {'master_user_name', 'master_user_password'}
>>> secret_data.keys() != required_keys
True

>>> secret_data = {'master_user_name': "u", 'master_user_password': "p"}
>>> required_keys = {'master_user_name', 'master_user_password'}
>>> secret_data.keys() != required_keys
False
```

Reply:
Fixed in 2e7228b